File: Editorial/58_Length_of_Last_Word/Go/2_One_loop_Iteration.go
func lengthOfLastWord(s string) int {
    p, length := len(s), 0
    for p > 0 {
        p--
        // we're in the middle of the last word
        if s[p:p+1] != " " {
            length++
        } else if length > 0 { // here is the end of last word
            return length
        }
    }
    return length
}

File: Editorial/58_Length_of_Last_Word/TypeScript/2_One_loop_Iteration.ts
function lengthOfLastWord(s: string): number {
    var p = s.length,
        length = 0;
    while (p > 0) {
        p--;
        // we're in the middle of the last word
        if (s[p] !== " ") {
            length++;
        }
        // here is the end of last word
        else if (length > 0) {
            return length;
        }
    }
    return length;
}
